src/NzbDrone.Core/HealthCheck/Checks/RemotePathMappingCheck.cs

```diff
@@ -0,0 +1,128 @@
+                    }
+                    else
+                    {
+                        // If the file doesn't exist but TrackInfo is not null then the message is coming from
+                        // ImportApprovedTracks and the file must have been removed part way through processing
+                        return new HealthCheck(GetType(), HealthCheckResult.Error, $"File {trackPath} was removed part way though procesing.");
+                    }
+                }
+
+                // If the previous case did not match then the failure occured in DownloadedTracksImportService,
+                // while trying to locate the files reported by the download client
+                var client = _downloadClientProvider.GetDownloadClients().FirstOrDefault(x => x.Definition.Name == failureMessage.DownloadClient);
+                var dlpath = client?.GetItems().FirstOrDefault(x => x.DownloadId == failureMessage.DownloadId)?.OutputPath.FullPath;
+
+                // If dlpath is null then there's not much useful we can report.  Give a generic message so
+                // that the user realises something is wrong.
+                if (dlpath.IsNullOrWhiteSpace())
+                {
+                    return new HealthCheck(GetType(), HealthCheckResult.Error, $"Lidarr failed to import a track.  Check your logs for details.");
+                }
+
+                if (_diskProvider.FolderExists(dlpath))
+                {
+                    return new HealthCheck(GetType(), HealthCheckResult.Error, $"Lidarr can see but not access download directory {dlpath}.  Likely permissions error.", "#permissions-error");
+                }
+                
+                // if it's a remote client/docker, likely missing path mappings
+                if (OsInfo.IsDocker)
+                {
+                    return new HealthCheck(GetType(), HealthCheckResult.Error, $"You are using docker; download client {client.Definition.Name} reported files in {dlpath} but this directory does not appear to exist inside the container.  Review your remote path mappings and container volume settings.", "#docker-bad-remote-path-mapping");
+                }
+                else if (!client.GetStatus().IsLocalhost)
+                {
+                    return new HealthCheck(GetType(), HealthCheckResult.Error, $"Remote download client {client.Definition.Name} reported files in {dlpath} but this directory does not appear to exist.  Likely missing remote path mapping.", "#bad-remote-path-mapping");
+                }
+                else
+                {
+                    // path mappings shouldn't be needed locally so probably a permissions issue
+                    return new HealthCheck(GetType(), HealthCheckResult.Error, $"Download client {client.Definition.Name} reported files in {dlpath} but Lidarr cannot see this directory.  You may need to adjust the folder's permissions.", "#permissions-error");
```

Comment:
Wont this also throw if folder really doesn't exist? should we modify the language `You may need to adjust the folder's permissions.`